src/write_hdf5_xml/write_hdf5_xml.c

```diff
@@ -2148,8 +2163,9 @@ int main(int argc, char **argv)
       " sensor\">%s</beam_mode>\n", meta->general->mode);
     fprintf(fp, "      <absolute_orbit type=\"int\" definition=\"absolute orbit"
       " of the image\">%d</absolute_orbit>\n", meta->general->orbit);
-    fprintf(fp, "      <frame type=\"int\" definition=\"frame number of the "
-      "image\">%d</frame>\n",  meta->general->frame);
+    if (strcmp_case(meta->general->sensor, "ALOS") == 0)
```

Comment:
Could this condition be:
if (meta->general->frame >= 0)
instead?  Is meta->general->frame set to MAGIC_UNSET_INT (which is -999... I think) by default?